doc/examples/applications/plot_3d_image_processing.py

```diff
@@ -0,0 +1,294 @@
+# Function `io.imshow` can only display grayscale and RGB(A) 2D images.
+# We can thus use it to visualize 2D planes. By fixing one axis, we can
+# observe three different views of the image.
+
+
+def show_plane(ax, plane, cmap="gray", title=None):
+    ax.imshow(plane, cmap=cmap)
+    ax.axis("off")
+
+    if title:
+        ax.set_title(title)
+
+
+_, (a, b, c) = plt.subplots(ncols=3, figsize=(15, 5))
+
+show_plane(a, data[32], title="Plane = 32")
+show_plane(b, data[:, 128, :], title="Row = 128")
+show_plane(c, data[:, :, 128], title="Column = 128")
+
+#####################################################################
+# As hinted before, a three-dimensional image can be viewed as a series of
+# two-dimensional planes. Let us write a helper function, `display`, to
+# display 30 planes of our data. By default, every other plane is displayed.
+
+
+def display(im3d, cmap="gray", step=2):
+    _, axes = plt.subplots(nrows=5, ncols=6, figsize=(16, 14))
+
+    vmin = im3d.min()
+    vmax = im3d.max()
+
+    for ax, image in zip(axes.flatten(), im3d[::step]):
+        ax.imshow(image, cmap=cmap, vmin=vmin, vmax=vmax)
+        ax.set_xticks([])
+        ax.set_yticks([])
+
+
+display(data)
+
+#####################################################################
+# Alternatively, we can explore these planes (slices) interactively using
+# Jupyter widgets. Let the user select which slice to display and show the
+# position of this slice in the 3D dataset.
+
+
+def slice_in_3D(ax, i):
+    # From https://stackoverflow.com/questions/44881885/python-draw-3d-cube
+    Z = np.array([[0, 0, 0],
+                  [1, 0, 0],
+                  [1, 1, 0],
+                  [0, 1, 0],
+                  [0, 0, 1],
+                  [1, 0, 1],
+                  [1, 1, 1],
+                  [0, 1, 1]])
+
+    Z = Z * data.shape
+    r = [-1, 1]
+    X, Y = np.meshgrid(r, r)
+
+    # Plot vertices
+    ax.scatter3D(Z[:, 0], Z[:, 1], Z[:, 2])
+
+    # List sides' polygons of figure
+    verts = [[Z[0], Z[1], Z[2], Z[3]],
+             [Z[4], Z[5], Z[6], Z[7]],
+             [Z[0], Z[1], Z[5], Z[4]],
+             [Z[2], Z[3], Z[7], Z[6]],
+             [Z[1], Z[2], Z[6], Z[5]],
+             [Z[4], Z[7], Z[3], Z[0]],
+             [Z[2], Z[3], Z[7], Z[6]]]
+
+    # Plot sides
+    ax.add_collection3d(
+        Poly3DCollection(
+            verts,
+            facecolors=(0, 1, 1, 0.25),
+            linewidths=1,
+            edgecolors="darkblue"
+        )
+    )
+
+    verts = np.array([[[0, 0, 0],
+                       [0, 0, 1],
+                       [0, 1, 1],
+                       [0, 1, 0]]])
+    verts = verts * (60, 256, 256)
+    verts += [i, 0, 0]
+
+    ax.add_collection3d(
+        Poly3DCollection(
+            verts,
+            facecolors="magenta",
+            linewidths=1,
+            edgecolors="black"
+        )
+    )
+
+    ax.set_xlabel("plane")
+    ax.set_ylabel("row")
+    ax.set_zlabel("col")
+
```

Comment:
```suggestion
ax.set_xlim(0, 100)

```
so that one sees better how the slice moves through the parallelepiped.